### ISSUE TITLE
RLHF第三步进行训练时，在modeling_glm.py如下代码位置会显示“size mismatch, got 1028, 1028x1024,0”

### ISSUE THREAD

**Issue by IconShan** [sunzeyeah/RLHF#14]
logits, hidden_layers = transformer_output
        # outputs = hidden_layers
        if self.output_predict:
            # Parallel logits.
            # logits_parallel = mpu.copy_to_model_parallel_region(
            #     logits)
            logits = F.linear(logits, self.word_embeddings.weight)

去modeling_glm.py中打印logits.size() 和self.word_embeddings.weight.size()时发现后者的shape为：torch.size([0])，
其中SFT和Reward的模型我在step1，step2中已经do_pred测试通过了，请问这种mismatch的问题有没有解决方案？感谢感谢~~

**Comment by sunzeyeah** [comment]
已验证方案2可以解决该问题，具体可参见[modeling_glm.py line 175-L176](https://github.com/sunzeyeah/RLHF/blob/master/src/resources/models/glm-350M-chinese/modeling_glm.py#L175-L176)

把modeling_glm.py文件复制到模型目录行替换即可

**Comment by IconShan** [comment]
感谢解答，已经跑通了，牛的牛的

**Comment by Bo396543018** [comment]
用上述方法2确实解决了这个问题。但是开启zero3不用hybrid_engine，glm10b在make_experience的时候结果是乱的，想问下你们有遇到类似问题吗。
